smtp/connection.ts

```diff
@@ -120,10 +120,7 @@ export class SMTPConnection extends EventEmitter {
 	protected tls: boolean | SMTPSocketOptions = false;
 	protected port: number;
 
-	private greylistResponseTracker = new WeakMap<
-		(...rest: any[]) => void,
-		boolean
-	>();
+	private greylistResponseTracker = new WeakSet<(...rest: any[]) => void>();
```

Comment:
made this change in anticipation of aligning with a sent message tracker in `SmtpClient`, only to discover the additional tracker wasn't necessary. at any rate, i think a `WeakSet` is more appropriate here anyway, so i left it in.